osutil/epoll/epoll_test.go

```diff
@@ -0,0 +1,546 @@
+package epoll_test
+
+import (
+	"os"
+	"syscall"
+	"testing"
+	"time"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/snapcore/snapd/osutil/epoll"
+
+	. "gopkg.in/check.v1"
+)
+
+func Test(t *testing.T) { TestingT(t) }
+
+type epollSuite struct{}
+
+var _ = Suite(&epollSuite{})
+
+func (*epollSuite) TestString(c *C) {
+	c.Check(epoll.Readable.String(), Equals, "Readable")
+	c.Check(epoll.Writable.String(), Equals, "Writable")
+	c.Check(epoll.Readiness(epoll.Readable|epoll.Writable).String(), Equals, "Readable|Writable")
+}
+
+func (*epollSuite) TestOpenClose(c *C) {
+	e, err := epoll.Open()
+	c.Assert(err, IsNil)
+	c.Assert(e.Fd() == -1, Equals, false)
+	c.Assert(e.Fd() == 0, Equals, false)
+	c.Assert(e.Fd() == 1, Equals, false)
+	c.Assert(e.Fd() == 2, Equals, false)
+	c.Assert(e.RegisteredFdCount(), Equals, 0)
+
+	err = e.Close()
+	c.Assert(err, IsNil)
+	c.Assert(e.Fd(), Equals, -1)
+}
+
+func concurrentlyRegister(e *epoll.Epoll, fd int, errCh chan error) {
+	err := e.Register(fd, epoll.Readable)
+	errCh <- err
+}
+
+func concurrentlyDeregister(e *epoll.Epoll, fd int, errCh chan error) {
+	err := e.Deregister(fd)
+	errCh <- err
+}
+
+func waitMillisecondsThenWriteToFd(msec int, fd int, msg []byte) error {
```

Comment:
(super nitpick) this could just be `waitThenWriteToFd(dur time.Duration, ...` and then we could have a single `defaultWait` for all the cases where we curerntly put 1000 maybe? This could be combined with testutil.HostScaledTimeout for the really slow arches (like riscv)

Reply:
In the timeout reduction commit I also made the change to use `time.Duration` here. I like the idea of using `testutil.HostScaledTimeout`... I think I'll make the `baseDuration := testutil.HostScaledTimeout(time.Millisecond * 50)` and then scale it by whatever is necessary to match the values I use right now. Thanks!

Reply:
I used 100ms base to keep things simpler.